src/main/java/io/apimatic/core/types/pagination/OffsetPagination.java

```diff
@@ -14,25 +17,36 @@ public OffsetPagination(final String input) {
     }
 
     @Override
-    public boolean isValid(PaginatedData<?, ?> paginatedData) {
-        nextReqBuilder = paginatedData.getLastRequestBuilder();
-
-        if (input == null) {
-            return false;
-        }
-
-        final boolean[] isUpdated = {false};
-        nextReqBuilder.updateByReference(input, old -> {
-            int newValue = Integer.parseInt("" + old) + paginatedData.getLastDataSize();
-            isUpdated[0] = true;
+    public Builder apply(PaginatedData<?, ?, ?, ?> paginatedData) {
+        Response response = paginatedData.getResponse();
+        Builder reqBuilder = paginatedData.getRequestBuilder();
+        AtomicBoolean isUpdated = new AtomicBoolean(false);
+
+        reqBuilder.updateParameterByJsonPointer(input, old -> {
```

Comment:
The offset and page based pagination should work till end if there are no offset/page parameters in the initial request by user.